test/kdump_test.rb

```diff
@@ -159,4 +167,290 @@
+        expect(several_ranges).to eq false
+      end
+
+      it "correctly reads the size" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+
+    context "when the param includes several ranges" do
+      let(:kernel_param) { "-200M:32M,200M-:64M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "finds several ranges" do
+        expect(several_ranges).to eq true
+      end
+
+      it "correctly reads the size of the last range" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+  end
+
+  describe ".WriteKdumpBootParameter" do
+    context "during autoinstallation" do
+      before do
+        allow(Yast::Mode).to receive(:autoinst).and_return true
+        allow(Yast::Mode).to receive(:update).and_return false
+        allow(Yast::Mode).to receive(:normal).and_return false
+        Yast::Kdump.Import(profile)
+      end
+
+      context "if kdump is requested and a value for crashkernel is supplied" do
+        let(:profile) { {"add_crash_kernel" => true, "crash_kernel" => "the_value"} }
+
+        it "writes the crashkernel value to the bootloader and enables the service" do
+          expect(Yast::Bootloader)
+            .to receive(:modify_kernel_params)
+            .with(:common, :xen_guest, :recovery, {"crashkernel" => "the_value"})
+          expect(Yast::Bootloader).to receive(:Write)
+          expect(Yast::Service).to receive(:Enable)
+
+          Yast::Kdump.WriteKdumpBootParameter
+        end
+      end
+
+      context "if kdump is requested but no value for crashkernel is supplied" do
+        let(:profile) { {"add_crash_kernel" => true} }
+
+        it "writes an empty crashkernel in the bootloader and enables the service" do
+          expect(Yast::Bootloader)
+            .to receive(:modify_kernel_params)
+            .with(:common, :xen_guest, :recovery, {"crashkernel" => ""})
+          expect(Yast::Bootloader).to receive(:Write)
+          expect(Yast::Service).to receive(:Enable)
+
+          Yast::Kdump.WriteKdumpBootParameter
+        end
+      end
+
+      context "if kdump is explicitly disabled" do
+        let(:profile) { {"add_crash_kernel" => false, "crash_kernel" => "does_not_matter"} }
+
+        it "disables the service not touching bootloader" do
+          allow(Yast::Service).to receive(:Status).and_return -1
+
+          expect(Yast::Bootloader).to_not receive(:modify_kernel_params)
+          expect(Yast::Bootloader).to_not receive(:Write)
+          expect(Yast::Service).to receive(:Disable)
+
+          Yast::Kdump.WriteKdumpBootParameter
+        end
+      end
+    end
+
+    context "during autoupdate of a system with kdump" do
+      before do
+        allow(Yast::Mode).to receive(:autoinst).and_return true
+        allow(Yast::Mode).to receive(:update).and_return false
```

Comment:
@imobach has run several autoinstallations for me and I have checked extensively the logs. `Read` is never used in the workflow of `Mode.autoinst`. So this whole context is testing something that will never happen (even if the Kdump code kinda "suggests" it can happen). Removing the context.

`Read`is surely used in the case of `Mode.autoupgrade`, but that's not relevant here because there is no special handling for `Mode.autoupgrade` in Kdump. So no test is needed.